src/Symfony/Component/Config/Definition/Builder/TreeBuilder.php

```diff
@@ -23,6 +23,15 @@ class TreeBuilder implements NodeParentInterface
     protected $tree;
     protected $root;
 
+    public function __construct(string $name = null, string $type = 'array', NodeBuilder $builder = null)
+    {
+        if (null === $name) {
+            @trigger_error('A tree builder without a root node is deprecated since Symfony 4.2 and will not be supported anymore with 5.0.', E_USER_DEPRECATED);
+        } else {
+            $this->root($name, $type, $builder);
```

Comment:
should we deprecate public calls to `root()` as well? or is it still intended as a feature?

Reply:
I do not really care to be honest. Since this is a builder there may be use cases to still call `root()` later. But I also see that we could probably discuss this as part of a follow up PR if you would like to open one. :)